components/support/migration/src/main/java/mozilla/components/support/migration/FennecProfile.kt

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.migration
+
+import android.content.Context
+import mozilla.components.support.base.log.logger.Logger
+import java.io.File
+import java.util.regex.Pattern
+
+private val logger = Logger("FennecProfile")
+private val profilePattern = Pattern.compile("\\[(.+)]")
+
+/**
+ * A profile of "Fennec" (Firefox for Android).
+ */
+data class FennecProfile(
+    val name: String,
+    val path: String,
+    val default: Boolean
+) {
+    companion object {
+        /**
+         * Returns the default [FennecProfile] - the default profile used by Fennec or `null` if no
+         * profile could be found.
+         *
+         * If no explicit default profile could be found then this method will try to return a valid
+         * profile that may be treated as the default profile:
+         * - If there's only one profile then this profile will be returned.
+         * - If there are multiple profiles with no default then the first profile will be returned.
+         */
+        fun findDefault(
+            context: Context,
+            mozillaDirectory: File = getMozillaDirectory(context),
+            fileName: String = "profiles.ini"
+        ): FennecProfile? {
+            return findDefaultProfile(mozillaDirectory, fileName)
+        }
+    }
+}
+
+private fun getMozillaDirectory(context: Context): File {
+    return File(context.filesDir, "mozilla")
+}
+
+@Suppress("ReturnCount")
+private fun findDefaultProfile(
+    mozillaDirectory: File,
+    fileName: String
+): FennecProfile? {
+    val profiles = findProfiles(mozillaDirectory, fileName)
+
+    if (profiles.isEmpty()) {
+        logger.debug("No profiles found")
+        return null
+    }
+
+    profiles.firstOrNull {
+        it.default
+    }?.let {
+        logger.debug("Found default profile: $it")
+        return it
+    }
+
+    if (profiles.size == 1) {
+        val profile = profiles[0]
+        logger.debug("No default found but using only existing profile:: $profile")
+        return profile
+    }
+
+    logger.debug("Multiple profiles without a default. Returning first.")
+    return profiles[0]
+}
+
+@Suppress("ComplexMethod")
+private fun findProfiles(
+    mozillaDirectory: File,
+    fileName: String = "profiles.ini"
+): List<FennecProfile> {
+    // Implementation note: Fennec saves information about its profiles in mozilla/profiles.ini.
+    // A Fennec installation can have multiple profiles and the profile names always contain a
+    // salt causing them to be different for every installation.
```

Comment:
🤔  I'm not quite sure what salt means in this context?

Reply:
https://en.wikipedia.org/wiki/Salt_(cryptography)

In this situation it means the folder for a profile named "arturo" will not be just  `arturo` but something like `10aaayu4-arturo`. `10aaayu4` will be the salt in this case.

Reply:
Awesome, thanks for the link! I didn't know that salt was useful outside the food business XD